### ISSUE TITLE
Remote ROS Master Restart

### ISSUE THREAD

**Issue by baronep** [clearpathrobotics/robot_upstart#35]
I have a laptop running roscore and I have a RasPi which is connected to a few cameras and connects to roscore on the laptop.

If roscore is not started, the RasPi seems to error-out and retry connecting to the remote roscore when configured with the --master option (good!), but if the roscore is restarted, the RasPi seems to fail silently and I have to restart the service. Ideally it should go into the polling mode if the roscore is stopped right?

**Comment by mikepurvis** [comment]
I agree with your expectations; I'd welcome a PR that corrects this, either by some in-job supervision scheme, or by exiting the job and relying on the underlying init system to restart the job and enter the waiting mode (for which we simply depend on roslaunch's ability to do this).
